packages/compiler/src/render3/r3_deferred_triggers.ts

```diff
@@ -141,27 +138,36 @@ class OnTriggerParser {
     try {
       switch (identifier.toString()) {
         case OnTriggerType.IDLE:
-          this.triggers.push(createIdleTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'idle', this.triggers, this.errors, createIdleTrigger(parameters, sourceSpan));
           break;
 
         case OnTriggerType.TIMER:
-          this.triggers.push(createTimerTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'timer', this.triggers, this.errors, createTimerTrigger(parameters, sourceSpan));
           break;
 
         case OnTriggerType.INTERACTION:
-          this.triggers.push(createInteractionTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'interaction', this.triggers, this.errors,
+              createInteractionTrigger(parameters, sourceSpan));
           break;
 
         case OnTriggerType.IMMEDIATE:
-          this.triggers.push(createImmediateTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'immediate', this.triggers, this.errors,
+              createImmediateTrigger(parameters, sourceSpan));
           break;
 
         case OnTriggerType.HOVER:
-          this.triggers.push(createHoverTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'hover', this.triggers, this.errors, createHoverTrigger(parameters, sourceSpan));
           break;
 
         case OnTriggerType.VIEWPORT:
-          this.triggers.push(createViewportTrigger(parameters, sourceSpan));
+          trackTrigger(
+              'viewport', this.triggers, this.errors,
+              createViewportTrigger(parameters, sourceSpan));
```

Comment:
Quick question: can we have a private function on this class instead?
```suggestion
          this.trackTrigger(
              'viewport', createViewportTrigger(parameters, sourceSpan));
```

Reply:
Good idea. Done.